sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_options.hpp

```diff
@@ -662,22 +662,18 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
      * @brief Specify the crc64 calculated for the range of bytes that must be read from the copy
      * source.
      */
-    Azure::Core::Nullable<ContentHash> SourceContentCrc64;
+    Azure::Core::Nullable<ContentHash> ContentHash;
 
     /**
-     * @brief Specify the crc64 value to operate only on range with a matching crc64 checksum.
+     * @brief Specify the access condition for the source. Only Crc64 is supported.
      */
-    Azure::Core::Nullable<ContentHash> SourceIfMatchCrc64;
+    ContentHashAccessConditions SourceContentHashAccessCondition;
+    Azure::Core::Nullable<std::string> SourceIfNoneMatchCrc64;
```

Comment:
Why isn't this in AccessCondition?

Reply:
no doc?

Reply:
why not combine `SourceContentHashAccessCondition` and `LeaseAccessConditions` into one struct and just call it `AccessConditions`, as aligned with other APIs?

Reply:
`Azure::Core::Nullable<std::string> SourceIfNoneMatchCrc64;` should be removed and `ContentHash` should be moved to `SourceContentHash`.
However, we should not combine `SourceContentHashAccessCondition` and `LeaseAccessConditions` as one is source access condition and one is destination access condition.

Reply:
> However, we should not combine SourceContentHashAccessCondition and LeaseAccessConditions as one is source access condition and one is destination access condition.

ah, I get it. So we should have `AccessCondition` and `SourceAccessCondition`.